src/transform.rs

```diff
@@ -0,0 +1,161 @@
+        Err(ZkError::BadVersion) => Ok(Err(error::Delete::BadVersion { expected: version })),
+        Err(e) => Err(format_err!("delete call failed: {:?}", e)),
+    }
+}
+
+pub(crate) fn get_acl(
+    res: Result<Response, ZkError>,
+) -> Result<Result<(Vec<Acl>, Stat), error::GetAcl>, failure::Error> {
+    match res {
+        Ok(Response::GetAcl { acl, stat }) => Ok(Ok((acl, stat))),
+        Ok(r) => bail!("got non-acl response to a get_acl request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(Err(error::GetAcl::NoNode)),
+        Err(e) => Err(format_err!("get_acl call failed: {:?}", e)),
+    }
+}
+
+pub(crate) fn set_acl(
+    version: i32,
+    res: Result<Response, ZkError>,
+) -> Result<Result<Stat, error::SetAcl>, failure::Error> {
+    match res {
+        Ok(Response::Stat(stat)) => Ok(Ok(stat)),
+        Ok(r) => bail!("got non-stat response to a set_acl request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(Err(error::SetAcl::NoNode)),
+        Err(ZkError::BadVersion) => Ok(Err(error::SetAcl::BadVersion { expected: version })),
+        Err(ZkError::InvalidACL) => Ok(Err(error::SetAcl::InvalidAcl)),
+        Err(ZkError::NoAuth) => Ok(Err(error::SetAcl::NoAuth)),
+        Err(e) => Err(format_err!("set_acl call failed: {:?}", e)),
+    }
+}
+
+pub(crate) fn exists(res: Result<Response, ZkError>) -> Result<Option<Stat>, failure::Error> {
+    match res {
+        Ok(Response::Stat(stat)) => Ok(Some(stat)),
+        Ok(r) => bail!("got a non-create response to a create request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(None),
+        Err(e) => bail!("exists call failed: {:?}", e),
+    }
+}
+
+pub(crate) fn get_children(res: Result<Response, ZkError>) -> Result<Option<Vec<String>>, failure::Error> {
+    match res {
+        Ok(Response::Strings(children)) => Ok(Some(children)),
+        Ok(r) => bail!("got non-strings response to get-children: {:?}", r),
+        Err(ZkError::NoNode) => Ok(None),
+        Err(e) => Err(format_err!("get-children call failed: {:?}", e)),
+    }
+}
+
+pub(crate) fn get_data(res: Result<Response, ZkError>) -> Result<Option<(Vec<u8>, Stat)>, failure::Error> {
+    match res {
+        Ok(Response::GetData { bytes, stat }) => Ok(Some((bytes, stat))),
+        Ok(r) => bail!("got non-data response to get-data: {:?}", r),
+        Err(ZkError::NoNode) => Ok(None),
+        Err(e) => Err(format_err!("get-data call failed: {:?}", e)),
+    }
+}
+
+pub(crate) fn check(
+    version: i32,
+    res: Result<Response, ZkError>,
+) -> Result<Result<(), error::Check>, failure::Error> {
+    match res {
+        Ok(Response::Empty) => Ok(Ok(())),
+        Ok(r) => bail!("got a non-check response to a check request: {:?}", r),
+        Err(ZkError::NoNode) => Ok(Err(error::Check::NoNode)),
+        Err(ZkError::BadVersion) => Ok(Err(error::Check::BadVersion { expected: version })),
+        Err(e) => bail!("check call failed: {:?}", e),
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum RequestMarker {
```

Comment:
I think I'd like to see a doc-comment for this that explains what it's used for (basically the outline of https://github.com/jonhoo/tokio-zookeeper/pull/15#discussion_r264320027)

Reply:
Done!